### ISSUE TITLE
We should set up a discord or discussion area outside of Stackoverflow

### ISSUE THREAD

**Issue by jeffward01** [ardalis/CleanArchitecture#281]
Hello all!

### **Proposal:**

I see that we can ask question on StackOverflow, which is excellent!  However, I suggest we open a website or discord server (or similar) where we can host open questions and answers.  

####  **The Why**

StackOverflow is amazing as we all know, yet stack overflow has very strict and sometimes unfair standards of questions. For example, if a question is asked that is considered unworthy, the question is quickly rejected and the stack overflow account which created the question gets a negative point towards its overall score. 

#### **What type of questions will be asked? Why is this helpful**

For example, I have some 'simple' questions such as:

* Why use AutoFac over the default Microsoft library?
* What is the best method to organize additional DTO's, Services, etc related to a single feature
* How and why to take advantage of putting interfaces in an isolated '.Contracts' project

etc

I realize we have Reddit for this, and other sources already, but given @ardalis size of following and number of very awesome libraries, it would be nice to have a dedicated discussion area for @ardalis styled C# development

**Comment by ardalis** [comment]
I like it; let me think about how this would work. Any Discord gurus want to help me with this?

**Comment by ardalis** [comment]
To be clear, GitHub supports per-repo Discussions now, but you're thinking something that would be a space for all of the Ardalis.* NuGet packages, perhaps with separate channels for each one but making it more of a community than just one-off tools or package users.

**Comment by jeffward01** [comment]
> To be clear, GitHub supports per-repo Discussions now, but you're thinking something that would be a space for all of the Ardalis.* NuGet packages, perhaps with separate channels for each one but making it more of a community than just one-off tools or package users.

I was thinking of something that would encompass all of 'Ardalis.* libraries, so it could serve as the homebase for 'building clean software the 'Adralis.* way'.

Users like me and other can ask questions about maybe things such as:

* Why in the clean code sample project is XX built like this
* In scenerio YY, am I using the Specs library correctly?
* I have some libraries (nuget packages) that use the 'Result' package, now i install this nuget package into my API solution, and add some logic for XYZ feature, is this the best way I can be doing this?
* Should I 'upgrade' Adrails.SmartEnum to records or sealed records instead of a sealed class

----

So this would not replace the individual discussions of each repository, but it would provide users with an area where architecture savvy people like to hang out, and learn, in order to write software in the most clean way possible, using best structures and best patterns overall.

Im sure in some cases, the answer would be "yes, that sounds like a good option to upgrade SmartEnum to records... open an issue on the github repo and we can continue the chat there" (100% example, idk if its a good or bad idea to change SmartEnum to records <3 )

This was what I was thinking to be honest.

My personal scenerio is I have some classes that are rather large, with managers that utilize services, and handlers, then seperate modules. I want to keep the code clean and vertically structured, so I would share my 'clean code example' in the discord (or whatever platform you choose) and people can say "Ya your a genius, I see why and how you do that, this is CleanCode 3.0!" or " No, I think this is overkill and not a good solution, you should do XYZ instead".

Basically a platform to encourage a good dialog and basic questions that are not a good fit for stackoverflow so that we can:

Build software and maintain clean code, the Ardalis way.

Hopefully this small rant cleared it up!

**Comment by jeffward01** [comment]
> I like it; let me think about how this would work. Any Discord gurus want to help me with this?

Im not exactly a discord guru, but i know a thing or two about discord and can help set it up! Im more limited on time rather than on skills.

I have seen some very configured discord chatrooms with ranking, speaking, auto-posting to github, etc. If another discord guru does not chime in, i can do this! I'm just a bit limited on time, however having a platform like this would be very cool in my opinion so i would love to dedicate some time to this!

**Comment by MisinformedDNA** [comment]
I vote for GitHub Discussions. I'm tired of going to multiple 3rd party sites to discuss repos (Discord, Gitter, Slack, etc.) Also, Discord is blocked in some companies. If having the discussions spread over different repos in a real concern, then we could use something like `ardalis/Discussions`.
